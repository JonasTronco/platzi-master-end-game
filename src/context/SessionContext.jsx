import { useState, useEffect, createContext, useMemo, useContext } from 'react';
import Router from 'next/router';

const SessionContext = createContext();
export function SessionProvider(props) {
  const [session, setSession] = useState(null);

  useEffect(() => {
    try {
      setSession(JSON.parse(localStorage.getItem('session')));
    } catch (error) {
      Router.push('/');
    }
  }, []);

  const value = useMemo(
    () => ({
      session,
    }),
    [session],
  );

  return <SessionContext.Provider value={value} {...props} />;
}

export function useSession() {
  const context = useContext(SessionContext);
  if (!context) {
    throw new Error('No Session Context Provider');
  }
  return context;
}
