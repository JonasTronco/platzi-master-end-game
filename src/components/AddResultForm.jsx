import PropTypes from 'prop-types';
import { useSession } from '../context/SessionContext';
import useRequest from '../hooks/useRequest';
import TemplateForm from './TemplateForm';

function AddResultForm({ orderId }) {
  const { session } = useSession();
  const { response, loading, error } = useRequest(
    session.token,
    `${process.env.NEXT_PUBLIC_API_URL}/orders/${orderId}`,
    1,
  );

  if (loading) {
    return <div className="loader" />;
  }

  if (error) {
    return <h3>{error.message}</h3>;
  }

  return (
    <div className="add-result">
      {response && response.data ? (
        <TemplateForm examName={response.data.shortName} />
      ) : (
        ''
      )}
    </div>
  );
}

AddResultForm.propTypes = {
  orderId: PropTypes.string.isRequired,
};

export default AddResultForm;
