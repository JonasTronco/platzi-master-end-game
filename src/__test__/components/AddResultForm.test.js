import AddResultForm from '../../components/AddResultForm';
import { create } from 'react-test-renderer';
import { mount } from 'enzyme';
import { SessionProvider } from '../../context/SessionContext';
describe('AddResultForm Component', () => {
  const obj = {
    session: {
      token: '',
      user: {
        id: '',
        username: '',
        typeOfUser: '',
        isActive: true,
        imageURL: '',
        firstName: '',
        lastName: '',
        defaultPath: ''
      }
    }
  };
  test('Should render AddResultForm component', () => {
    const component = create(
      <SessionProvider value={obj}>
        <AddResultForm userID={''} orderId={''} />
      </SessionProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });
  it('<AddResultForm />', () => {
    const component = mount(
      <SessionProvider value={obj}>
        <AddResultForm userID={''} orderId={''} />
      </SessionProvider>
    );
    expect(component.length).toEqual(1);
  });
});
