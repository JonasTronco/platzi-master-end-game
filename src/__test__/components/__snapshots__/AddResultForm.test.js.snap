// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddResultForm Component Should render AddResultForm component 1`] = `
<div
  className="add-result"
>
  
</div>
`;
